### PR TITLE
[WIP](struct-type) support struct-type in vectorize engine

### LINKED ISSUE #14917: [Proposal] Support Struct Type
### Description

DSIP: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=231118769

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xy720**
# Proposed changes

Issue Number: close #14917 

## Problem summary

Work in progress.

This CL includes:
1、Change the be column meta to support struct type.(finish)
2、Add ColumnReader and ColumnWriter to support struct type. (finish)
3、Support creating table sql. (finish)

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -49,6 +50,23 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ColumnMetaPB&
...
+        case FieldType::OLAP_FIELD_TYPE_STRUCT: {
+            // not support empty struct
+            DCHECK(meta.children_columns_size() >= 1);
+            // create struct column reader
+            std::unique_ptr<ColumnReader> struct_reader(
+                    new ColumnReader(opts, meta, num_rows, file_reader));
+            struct_reader->_sub_readers.resize(meta.children_columns_size());
+            for (size_t i = 0; i < meta.children_columns_size(); i++) {
+                std::unique_ptr<ColumnReader> sub_reader;
+                RETURN_IF_ERROR(ColumnReader::create(opts, meta.children_columns(i),
```
Comment (cambyzju) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:62]: for loop may return early, it is better not use resize, use reserve instead.

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 2]
```diff
@@ -398,6 +416,21 @@ Status ColumnReader::new_iterator(ColumnIterator** iterator) {
     } else {
         auto type = (FieldType)_meta.type();
         switch (type) {
+        case FieldType::OLAP_FIELD_TYPE_STRUCT: {
+            std::vector<ColumnIterator*> sub_column_iterators;
+            size_t child_size = is_nullable() ? _sub_readers.size() - 1 : _sub_readers.size();
+            sub_column_iterators.resize(child_size);
```
Comment (cambyzju) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:422]: use reserve

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 3]
```diff
@@ -695,5 +869,7 @@ Status ArrayColumnWriter::finish_current_page() {
     return Status::NotSupported("array writer has no data, can not finish_current_page");
 }
 
+
```
Comment (cambyzju) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:872]: duplicate empty lines

File: be/src/olap/rowset/segment_v2/column_reader.cpp (hunk 1 above)
Comment (xy720) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:62]: I will fix it.

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**